### PR TITLE
[Neutron] Update deployment config

### PR DESCRIPTION
**Pull request by pacmanifold**

### CODE REVIEW

File: scripts/deploy/neutron/config_mainnet.ts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    'bundle bundle orchard jeans office umbrella bird around taxi arrive infant discover elder they joy misery photo crunch gift fancy pledge attend adult eight', // TODO: Set mnemonic before deploying
+  multisigAddr: 'TODO',
+  slippage_tolerance: '0.01',
+  base_asset_symbol: 'NTRN',
+  second_asset_symbol: 'ATOM',
+  runTests: false,
+  mainnet: true,
+  feeCollectorDenom: marsDenom,
+  safetyFundDenom: axlUsdcDenom,
+  swapRoutes: [atomNtrnRoute, atomUsdcRoute, ntrnUsdcRoute, atomMarsRoute],
```
Comment (piobab) [inline scripts/deploy/neutron/config_mainnet.ts:290]: @markonmars could you please double check the routes?

**Review by dancreee — APPROVED** [review]

**Review by dancreee — CHANGES_REQUESTED** [review]
see comments on TWAP price sources should be normalised to USD

File: scripts/deploy/neutron/config_mainnet.ts [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      price_feed_id: pythUsdcID,
+      denom_decimals: 6,
+      max_staleness: 60,
+    },
+  },
+}
+
+export const marsOracle: OracleConfig = {
+  denom: marsDenom,
+  price_source: {
```
[the remark this answers is not in the document]
Comment (dancreee) [reply scripts/deploy/neutron/config_mainnet.ts:85]: Do we need MARS in the oracle as a price source I mean*

File: scripts/deploy/neutron/config_mainnet.ts [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+      amount: '1000',
+    },
+  ],
+}
+
+// Oracle configurations
+export const ntrnOracle: OracleConfig = {
+  denom: 'untrn',
+  price_source: {
+    astroport_twap: {
```
[the remark this answers is not in the document]
Comment (dancreee) [reply scripts/deploy/neutron/config_mainnet.ts:51]: here is the NTRN/ATOM pair: https://www.mintscan.io/neutron/wasm/contract/neutron1e22zh5p8meddxjclevuhjmfj69jxfsa8uu3jvht72rv9d8lkhves6t8veq

[the remark this answers is not in the document]
Comment (piobab) [reply scripts/deploy/neutron/config_mainnet.ts:51]: Changed to Ntrn <-> Atom pool.
If second asset (in that case atom != uusd) we take price source for the atom and multiply by first one. See https://github.com/mars-protocol/red-bank/blob/master/contracts/oracle/wasm/src/helpers.rs#L163

File: scripts/deploy/neutron/config_mainnet.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (piobab) [reply scripts/deploy/neutron/config_mainnet.ts]: Confirmed with Tim. Removed.

File: scripts/deploy/neutron/config_mainnet.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    oracle: '', // Will be filled in by deploy script
+    router: astroportRouter,
+  },
+}
+
+// Asset configurations
+export const ntrnAsset: AssetConfig = {
+  denom: 'untrn',
+  max_loan_to_value: '0.35',
+  reserve_factor: '0.1',
```
Comment (piobab) [inline scripts/deploy/neutron/config_mainnet.ts:201]: Changed to 0.1 for all assets. Confirmed with Jon.

**Review by dancreee — APPROVED** [review]

File: scripts/deploy/neutron/config_mainnet.ts [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+      contract_addr: pythAddr,
+      price_feed_id: pythUsdcID,
+      denom_decimals: 6,
+      max_staleness: 60,
+    },
+  },
+}
+
+export const usdOracle: OracleConfig = {
+  denom: axlUsdcDenom,
```
Comment (dancreee) [inline scripts/deploy/neutron/config_mainnet.ts:84]: I think this denom should be 'usd' not 'ibc/F082B65C88E4B6D5EF1DB243CDA1D331D002759E938A0F5CD3FFDC5D53B3E349'?

**Review by dancreee — CHANGES_REQUESTED** [review]

File: scripts/deploy/neutron/config_mainnet.ts [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+  price_source: {
+    fixed: {
+      price: '1000000',
+    },
+  },
+}
+
+// Router configurations
+export const atomNtrnRoute = {
```
Comment (dancreee) [inline scripts/deploy/neutron/config_mainnet.ts:93]: not sure why we need an atom -> ntrn route? should only need atom -> usdc.axl + atom -> mars?

File: scripts/deploy/neutron/config_mainnet.ts [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+          },
+        },
+      },
+    ],
+    oracle: '', // Will be filled in by deploy script
+    router: astroportRouter,
+  },
+}
+
+export const ntrnUsdcRoute = {
```
Comment (dancreee) [inline scripts/deploy/neutron/config_mainnet.ts:145]: we seem to be missing

1. ntrn -> mars route
2. usdc.axl -> mars route

Essentially for every market asset we need a route to both usdc.axl and mars - the only asset we don't need usdc.axl route for is usdc.axl asset itself (but that asset still needs a mars route)

**Review by dancreee — CHANGES_REQUESTED** [review]

**Comment by dancreee** [comment]
@markonmars could you assist in reviewing the swap routes for rewards collector please?

File: scripts/deploy/neutron/config_mainnet.ts
Comment (markonmars) [inline scripts/deploy/neutron/config_mainnet.ts:1]: We need to have some additional routes for us to be able to liquidate properly. 

Current status of what IMO are required routes based on this config : 

- [x] atom -> mars
- [ ] atom -> ntrn
- [x] atom -> usdc

- [x] ntrn -> mars
- [ ] ntrn -> atom
- [x] ntrn -> usdc

- [ ] usdc -> atom
- [x] usdc -> mars
- [ ] usdc -> ntrn

Comment (markonmars) [reply scripts/deploy/neutron/config_mainnet.ts:1]: You're right it does, so we will need those as well.

File: scripts/deploy/neutron/config_mainnet.ts (hunk 1 above)
Comment (markonmars) [reply scripts/deploy/neutron/config_mainnet.ts]: added my thoughts [here](https://github.com/mars-protocol/red-bank/pull/264#discussion_r1278688621)

Comment (markonmars) [reply scripts/deploy/neutron/config_mainnet.ts]: LGTM

**Comment by piobab** [comment]
Updated testnet addresses to the one on neutron testnet.

**Review by piobab — APPROVED** [review]

**Comment by piobab** [comment]
Before merge to master all commits should be squashed into one.

**Review by markonmars — APPROVED** [review]

**Comment by piobab** [comment]
Squashed to single commit. Diff between old https://github.com/mars-protocol/red-bank/tree/feat/update-deployment-config-copy and current one is empty so should be good.

**Pull request merged**